### PR TITLE
feat: improve and fix data providers

### PR DESCRIPTION
**Pull request by MartinGbz**
**TL;DR: this PR fix some issue we had on a few Data Provider functions, add something as value (balance, tokenId, creation date...) for the group (and not just a defaultValue), and also remove useless providers.**

Details:
- **Alchemy Provider:**
  - feat: NFT Balance as value for getOwnersForCollection() function
  - fix: getOwnersOfTokenIds function (wasn’t working anymore) + put tokenId as value (not possible to have the balance for this one)
- **Ankr provider:**
  - feat: put balance as values for getTokenHolders (ERC-20 holders)  (not possible to have other things as value for NFT holders function)
- **DegenScore Provider:**
  - feat: put the degen score as value instead of the minimal score specified
  - Finally decided to remove it from factory because it’s just a fetch of onchain ERC-1155, so I we can fetch them using the token provider
  - Also no need to have a data provider for this: we don’t want to generate sub group of degenscore address, we want only 1 (source of truth) group and generate zk-proof from it.
  - So I created a new maintained group degenscore-beacon-holders (using the token provider)
- **ENS Subdomains Provider:**
  - Put the number of subdomain hold as value
- **Ethleaderboard Provider:** just un update here: like hive it’s also not updated anymore because of twitter API
- **Galxe Provider:**
  - fix: there was a deprecated attribute use in the request so I updated it
  - feat: put the tokenId as value (couldn’t put the balance unfortunately)
  - feat: created a new maintained group galxe-eth-merge-day (https://galxe.com/taiko/campaign/GCyruUtEW4) (this will replace the current: « exemple galxe » group). it allows us to have an maintained group for this Data Provider
- **Guild Provider:**
  - feat: put the roleId as value (I updated the description of the maintained group that use this provider)
- **Otterspace Provider:**
  - fix: the provider because it wasn’t working anymore
  - feat: add multiple networks (not only optimism)
  - feat: put creationDate of the badge as value (it’s the only interesting thing I could to implement)
  - I’ll also create a maintained group after the merge of this PR (so we have an example for this data provider too)
- **Unlock Provider:**
  - feat: put tokenId as value

### CODE REVIEW

File: group-generators/helpers/data-providers/alchemy/index.ts [hunk 1]
```diff
@@ -99,23 +100,23 @@ export class AlchemyProvider {
...
-      for (const owner of owners) {
-        groupData[owner] = 1;
-      }
+      groupData = { ...groupData, ...owners };
     }
     return groupData;
   }
 
-  private async _getOwnersOfTokenIds(tokenId: string) {
+  private async _getOwnersOfTokenIds(tokenId: string): Promise<FetchedData> {
```
[the remark this answers is not in the document]
Comment (MartinGbz) [reply group-generators/helpers/data-providers/alchemy/index.ts:111]: Yes I agree with you, the function has been created to be only used by `getOwnersOfTokenIds`, that is why it keep the same name. But I agree it's more logical to have `_getOwnersOfTokenId` 👍
